src/utils/ddb/ddb_commands.c

```diff
@@ -1275,3 +1283,170 @@ ddb_run_dev_replace(struct ddb_ctx *ctx, struct dev_replace_options *opt)
+	int             rc;
+
+	rc = d_hlc2timespec(epoch, &tspec);
+	if (!SUCCESS(rc)) {
+		ddb_errorf(ctx, "Conversion error of the DTX stat %s: " DF_RC "\n", stat_name,
+			   DP_RC(rc));
+		return rc;
+	}
+	rc = timespec2str(&tspec, buf, TIMESPEC_FMT_SIZE);
+	if (!SUCCESS(rc)) {
+		ddb_errorf(ctx, "Conversion error of the DTX stat %s: " DF_RC "\n", stat_name,
+			   DP_RC(rc));
+		return rc;
+	}
+	ddb_printf(ctx, "\t- %s time:%s%s, %" PRIu64 "\n", stat_name, align, buf, epoch);
+
+	return -DER_SUCCESS;
+}
+
+struct dtx_stat_args {
+	struct ddb_ctx          *ctx;
+	struct dtx_stat_options *opt;
+	uint32_t                 cont_seen;
+	uint32_t                 cmt_cnt;
+};
+
+static int
+dtx_stat(struct ddb_ctx *ctx, struct dv_indexed_tree_path *itp, uint32_t *cnt)
+{
+	struct dtx_stat stat;
+	daos_handle_t   coh = {0};
+	uint32_t        cnt_tmp;
+	int             rc;
+
+	rc = dv_cont_open(ctx->dc_poh, itp_cont(itp), &coh);
+	if (!SUCCESS(rc))
+		goto done;
+
+	cnt_tmp = vos_dtx_get_cmt_cnt(coh);
+	vos_dtx_stat(coh, &stat, 0);
+
+	ddb_print(ctx, "DTX entries statistics of ");
+	itp_print_full(ctx, itp);
+	ddb_print(ctx, "\n");
+	ddb_printf(ctx, "\t- Number of committed DTX of the container:\t%" PRIu32 "\n", cnt_tmp);
+	rc =
+	    dtx_print_time_stat(ctx, "DTX newest aggregated", stat.dtx_newest_aggregated, "\t\t\t");
+	if (!SUCCESS(rc))
+		goto done;
+	if (cnt != NULL)
+		*cnt = cnt_tmp;
+done:
+	dv_cont_close(&coh);
+
+	return rc;
+}
+
+static int
+dtx_stat_cb(daos_handle_t ih, vos_iter_entry_t *entry, vos_iter_type_t type,
+	    vos_iter_param_t *param, void *cb_arg, unsigned int *acts)
+{
+	struct ddb_ctx             *ctx;
+	struct dv_indexed_tree_path itp = {0};
+	struct dtx_stat_args       *args;
+	uint32_t                    cnt;
+	int                         rc;
+
+	if (type != VOS_ITER_COUUID) {
+		rc = -DER_SUCCESS;
+		goto done;
+	}
+
+	args = (struct dtx_stat_args *)cb_arg;
+	ctx  = args->ctx;
+
+	itp_set_cont(&itp, entry->ie_couuid, args->cont_seen);
+	++args->cont_seen;
+
+	rc = dtx_stat(ctx, &itp, &cnt);
+	if (!SUCCESS(rc))
+		goto done;
+	args->cmt_cnt += cnt;
+
+done:
+	itp_free(&itp);
+
+	return rc;
+}
+
+int
+ddb_run_dtx_stat(struct ddb_ctx *ctx, struct dtx_stat_options *opt)
+{
+	vos_iter_param_t     param = {0};
+	struct dtx_stat_args args  = {0};
+	int                  rc;
+
+	if (daos_handle_is_inval(ctx->dc_poh)) {
+		ddb_error(ctx, "Not connected to a pool. Use 'open' to connect to a pool.\n");
+		return -DER_NONEXIST;
+	}
+
+	if (opt->path != NULL && opt->path[0] != '\0') {
+		struct dv_indexed_tree_path itp = {0};
+
+		rc = init_path(ctx, opt->path, &itp);
+		if (!SUCCESS(rc))
+			goto done;
+
+		if (!itp_has_cont(&itp)) {
+			rc = -DER_INVAL;
+			goto done;
+		}
+
+		rc = dtx_stat(ctx, &itp, NULL);
+done:
+		itp_free(&itp);
+		return rc;
+	}
+
+	args.ctx            = ctx;
+	args.opt            = opt;
+	param.ip_hdl        = ctx->dc_poh;
+	param.ip_epr.epr_hi = DAOS_EPOCH_MAX;
+	do {
+		struct vos_iter_anchors anchors = {0};
```

Comment:
Move `anchors` outside the `while()` block; otherwise, you will reset the iteration repeatedly and fall into dead loop.

Reply:
> Arghhhh, not yet used to manage this kind of iterator.
> 
> * [x]  Fix iterator management.

Fixed with commit 49be49554